query-overview.md

```diff
@@ -0,0 +1,122 @@
+query only ever returns one value for each requested output feature, which
+it retrieves either from the cache or by executing resolvers.
+
+An offline query returns all requested computed features for all requested ids.
+For instance, say you've computed the number of transactions (`num_transactions`)
+for a User with id 1 every day over the past week. Making an offline query
+with `user.id=1` as an input and `user.num_transactions` as an output would
+return all seven of the previously computed feature `num_transaction` feature
+values.
+
+Offline queries are often restricted by lower and upper bound timestamps
+which constrain the outputs to a specific temporal range.
+
+## Query Side Effects
+
+Chalk queries return and write data. This is an essential part of
+Chalk: every time you compute a feature, either through an online or offline
+query, the output is written down in the offline store. This makes it easy to:
+
+- create datasets from your previously computed features,
+- monitor and track your computed features over time.
+
+| Online query                                                                               | Offline query                                                                                                                             |
+| ------------------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------- |
+| Returns one row of data about one entity                                                   | Returns a DataFrame of many rows of historical data corresponding to multiple entities point-in-time                                      |
+| Designed to return data immediately in milliseconds                                        | Blocks until computation is complete, not designed for millisecond-level computation                                                      |
+| Queries the online store, which caches recent data from online queries for quick retrieval | Queries the offline store (Timescale), which stores all data from both online and offline queries                                         |
+| Writes output data to online store database and offline store database                     | Writes output to offline store database and a parquet file containing results to cloud storage. Only writes to online store if specified. |
+
+---
+
+## Running Queries
+
+Queries are executed against deployments or branches. The easiest way to execute
+a query is either: 1). with the Chalk CLI, or 2). with one of Chalk's API
+clients.
+
+### Running Online Queries
+
+Online Queries can be run using the chalk CLI:
+
+```bash
+chalk query --in user.id=1 --in user.name=mary --out user
+```
+
+or one of our API Clients:
+
+```python
+from chalk.client import ChalkClient
+
+client = ChalkClient()
+client.query(
+  inputy={'user.id': 1},
```

Comment:
might want to make this query equivalent to the CLI query above for consistency?